components/address-input.jsx

```diff
@@ -637,4 +656,77 @@ var AddressInput = React.createClass({
   }
 });
 
-module.exports = AddressInput;
+var PartialAddress = React.createClass({
+  mixins: [IntlMixin, require("../mixins/input.jsx")],
+  propTypes: {
+    error: React.PropTypes.object.isRequired,
+    onChange: React.PropTypes.func.isRequired
+  },
+  getInitialState: function() {
+    return {
+      values: {
+        country: "US",
+        code: ""
+      },
+      countryValid: true,
+      codeValid: true
+    };
+  },
+  onCountryChange: function(e) {
+    this.onInput("country", e.currentTarget.value);
+  },
+  onCodeChange: function(e) {
+    this.onInput("code", e.currentTarget.value);
+  },
+  validate: function() {
+    var valid = this.validateFields(["code"]);
+    if (!this.refs.countrySelect.validate()) {
+      valid = false;
+      this.setState({
+        countryValid: valid
+      });
+    }
+    return valid;
+  },
+  render: function() {
+    var countryClassName = "";
+    if (!this.state.countryValid) {
+      countryClassName += " parsley-error";
+    }
+    var errorMessageClassName = "row error-msg-row";
+    var errorMessage = this.props.error.code;
+    var codeClassName = "";
+    if (!this.state.codeValid || errorMessage) {
+      codeClassName += "parsley-error";
+    }
+    if (errorMessage === "") {
+      errorMessageClassName += " hidden";
+    }
+    return (
+      <div className="row address-input">
+        <div className="full">
+          <CountrySelect ref="countrySelect" country={this.state.values.country} onCountryChange={this.onCountryChange} className={countryClassName}/>
+        </div>
+        <div className="full">
+          <input className={codeClassName} onChange={this.onCodeChange} value={this.state.values.code} type="text" name="zip" placeholder={this.getIntlMessage('postal_code')}/>
+        </div>
+        <div className={errorMessageClassName}>
+          <div className="full">
+            <div id="amount-error-msg">
+              <ul id="parsley-id-multiple-donation_amount" className="parsley-errors-list filled">
+                <li className="parsley-custom-error-message">
+                  {errorMessage}
+                </li>
+              </ul>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+});
+
+module.exports = {
+  FullAddress: FullAddress,
+  PartialAddress: PartialAddress
```

Comment:
Returning the two types of address, I didn't get this to work with an array, the blocks themselves are not boiled down enough, so I didn't bother.